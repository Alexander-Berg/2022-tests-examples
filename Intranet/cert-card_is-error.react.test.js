import React from 'react';
import CertCard from 'b:cert-card m:is-error';
import {mount} from 'enzyme';

it('should contain a proper content', () => {
    const wrapper = mount(
        <CertCard
            is-error
            fetchError={new Error('Error')}
            onCloseClick={jest.fn()}
            onRetryClick={jest.fn()}
            cert={{
                serial_number: 'serial-number',
                common_name: 'common name',
                ca_name: 'ca-name',
                type_human: {ru: 'type'},
                status_human: {ru: 'status'},
                device_platform: 'platform',
                end_date: '2018-06-28T19:37:53+03:00'
            }}
        />
    );

    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
});
