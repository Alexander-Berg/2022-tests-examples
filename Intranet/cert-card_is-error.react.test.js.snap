// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should contain a proper content 1`] = `
<cert-card
  cert={
    Object {
      "ca_name": "ca-name",
      "common_name": "common name",
      "device_platform": "platform",
      "end_date": "2018-06-28T19:37:53+03:00",
      "serial_number": "serial-number",
      "status_human": Object {
        "ru": "status",
      },
      "type_human": Object {
        "ru": "type",
      },
    }
  }
  fetchError={[Error: Error]}
  is-error={true}
  onCloseClick={[MockFunction]}
  onRetryClick={[MockFunction]}
>
  <div
    className="cert-card cert-card_is-error"
    data-lego="react"
    style={Object {}}
  >
    <cert-card__title
      key="title"
      text="Произошла ошибка"
    >
      <div
        className="cert-card__title"
        data-lego="react"
        style={Object {}}
      >
        Произошла ошибка
      </div>
    </cert-card__title>
    <error-text
      key="error-text"
    >
      <div
        className="error-text"
        data-lego="react"
        style={Object {}}
      >
        {}
      </div>
    </error-text>
    <div
      key="controls"
    >
      <button2
        mix={
          Array [
            Object {
              "block": "cert-card",
              "elem": "button",
              "mods": Object {},
            },
            Object {
              "block": "cert-card",
              "elem": "button-retry",
              "mods": Object {},
            },
          ]
        }
        onClick={[MockFunction]}
        size="s"
        tabIndex={0}
        theme="action"
        tone="default"
        view="default"
      >
        <button
          aria-disabled="false"
          aria-pressed="false"
          autoComplete="off"
          className="control button2 button2_view_default button2_tone_default button2_size_s button2_theme_action cert-card__button cert-card__button-retry"
          data-lego="react"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          <button2__text
            key="text"
          >
            <span
              className="button2__text"
              data-lego="react"
              style={Object {}}
            >
              Попробовать ещё раз
            </span>
          </button2__text>
        </button>
      </button2>
      <button2
        mix={
          Object {
            "block": "cert-card",
            "elem": "button",
            "mods": Object {},
          }
        }
        onClick={[MockFunction]}
        size="s"
        tabIndex={0}
        theme="normal"
        tone="default"
        view="default"
      >
        <button
          aria-disabled="false"
          aria-pressed="false"
          autoComplete="off"
          className="control button2 button2_view_default button2_tone_default button2_size_s button2_theme_normal cert-card__button"
          data-lego="react"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          <button2__text
            key="text"
          >
            <span
              className="button2__text"
              data-lego="react"
              style={Object {}}
            >
              Закрыть
            </span>
          </button2__text>
        </button>
      </button2>
    </div>
  </div>
</cert-card>
`;
